snuba/subscriptions/scheduler_processing_strategy.py

```diff
@@ -151,3 +165,72 @@ def terminate(self) -> None:
     def join(self, timeout: Optional[float] = None) -> None:
         self.__next_step.close()
         self.__next_step.join(timeout)
+
+
+class ScheduleSubscriptions(ProcessingStrategy[CommittableTick]):
+    """"
+    This strategy is responsible for scheduling all of the subscriptions
+    for a given tick.
+
+    The subscriptions to be scheduled are those that:
+    - Have the same partition index as the tick received, and
+    - Are scheduled to be run within the time interval defined by the tick
+
+    For backward compatibility, this assumes that the number of partitions of
+    the subscription storage matches the number of partitions of the main topic
+    being subscribed to.
```

Comment:
What do we gain by keeping the storage partitioned ini the same way as the topic partitioning?
It seem coupling the two is a limitation as there is no safe way to increase the number of topic partition without any sort of outage.

Can we keep track here of the last timestamp we scheduled, and as soon as we see a tick with an upper timestamp that is higher than the watermark, schedule all the subscriptions across partitions ?

Reply:
I removed this comment - it wasn't actually accurate, and there is no need for the partitioning strategies to be the same.